packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1235,13 +1235,23 @@ mixin TextSelectionDelegate {
   ///   updating the text editing state.
   void userUpdateTextEditingValue(TextEditingValue value, SelectionChangedCause cause);
 
+  /// Shows the text selection toolbar.
+  void showToolbar();
+
   /// Hides the text selection toolbar.
   ///
   /// By default, hideHandles is true, and the toolbar is hidden along with its
   /// handles. If hideHandles is set to false, then the toolbar will be hidden
   /// but the handles will remain.
   void hideToolbar([bool hideHandles = true]);
 
+  /// Toggles the visibility of the toolbar.
+  ///
+  /// If the toolbar is hidden calling this method should show the toolbar.
+  ///
+  /// If the toolbar is visible calling this method should hide the toolbar.
```

Comment:
I'd consider mentioning that the handles will remain visible if they are visible.

Reply:
Good call!